yascheduler/clouds/upcloud.py

```diff
@@ -50,7 +50,8 @@ def upcloud_create_node_sync(
             user_data=cloud_config.render() if cloud_config else None,
         )
     )
-    ip_addr = server.get_public_ip()
+    ip_addr = cast(Optional[str], server.get_public_ip())
```

Comment:
wow, a modern Python becomes so complex!

Reply:
It's just a combination of a shitty library and my "I don't want to deal with UpCloud"